src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -666,13 +670,21 @@
   (-plan->sql [query]
     (let [has-where? (boolean (:where query))
           has-projections? (not (empty? (:projected-fields query)))
-          update-when (fn [m pred ks f]
-                        (if pred
-                          (update-in m ks f)
-                          m))
+          update-hash-fields
```

Comment:
Would this lower the amount of changes in general for this patch, if we got this working?